src/components/KeyboardAvoidingView/index.ios.js

```diff
@@ -1,13 +1,74 @@
 /*
  * The KeyboardAvoidingView is only used on ios
  */
-import React from 'react';
-import {KeyboardAvoidingView as KeyboardAvoidingViewComponent} from 'react-native';
+import React, {useMemo} from 'react';
+import {
+    StyleSheet,
+    KeyboardAvoidingView as RNKeyboardAvoidingView,
+} from 'react-native';
+import Reanimated, {KeyboardState, useAnimatedKeyboard, useAnimatedStyle} from 'react-native-reanimated';
+import {makeRemote} from 'react-native-reanimated/src/reanimated2/core';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
 
-const KeyboardAvoidingView = props => (
-    // eslint-disable-next-line react/jsx-props-no-spreading
-    <KeyboardAvoidingViewComponent {...props} />
-);
+const KeyboardAvoidingView = (props) => {
+    const behavior = props.behavior;
+
+    const keyboard = useAnimatedKeyboard();
+    const insets = useSafeAreaInsets();
+
+    // makeRemote makes an object that can be used on the UI thread
+    // similar to using `global` in worklet but it's just a local object
+    const ctx = useMemo(() => makeRemote({}), []);
```

Comment:
I feel a bit suspicious that we need to use `makeRemote`. It's not included anywhere in their public documentation, and is not included in the main exports of the library.